### PR TITLE
Propagate capabilities to next index

### PR DESCRIPTION
**Pull request by navis**
column names are propagated but capabilities are not. just a refactoring for consistency.

### CODE REVIEW

**Comment by xvrl** [comment]
@navis we should add tests for this.

File: processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java [hunk 1]
```diff
@@ -800,19 +800,25 @@ public DimDim getDimensionValues(String dimension)
    * Index dimension ordering could be changed to initialize from DimensionsSpec after resolution of
    * https://github.com/druid-io/druid/issues/2011
    */
-  public void loadDimensionIterable(Iterable<String> oldDimensionOrder)
+  public void loadDimensionIterable(Map<String, ColumnCapabilities> oldDimensionOrder)
```
Comment (nishantmonu51) [inline processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java]: minor nit: rename method to loadDimensions

**Comment by nishantmonu51** [comment]
looks good to me, please add tests.

**Comment by navis** [comment]
@xvrl @nishantmonu51 I tried but cannot make some meaningful tests, not yet because all dimensions are still string type. I think we should wait for progress of numeric type dimensions.

**Comment by navis** [comment]
waiting #2760

**Pull request closed without merging**